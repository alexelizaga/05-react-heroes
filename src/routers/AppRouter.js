import { Routes, Route, BrowserRouter } from 'react-router-dom';
import { DcScreen } from '../components/dc/DcScreen';
import { LoginScreen } from '../components/login/LoginScreen';

import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';
import { MyNavbar } from '../components/ui/MyNavbar';

export const AppRouter = () => {
  return (
    <BrowserRouter>
      <MyNavbar />
      <Routes>
        <Route path='/' element={<MarvelScreen />} />
        <Route path='/marvel' element={<MarvelScreen />} />
        <Route path='/dc' element={<DcScreen />} />
        <Route path='/search' element={<SearchScreen />} />
        <Route path='/login' element={<LoginScreen />} />
      </Routes>
    </BrowserRouter>
  )
}
