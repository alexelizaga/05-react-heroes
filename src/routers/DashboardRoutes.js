import { Routes, Route } from 'react-router-dom';

import { MyNavbar } from '../components/ui/MyNavbar'
import { DcScreen } from '../components/dc/DcScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';
import { HeroScreen } from '../components/hero/HeroScreen';

export const DashboardRoutes = () => {
  return (
    <>
      <MyNavbar />

      <div className='container'>
        <Routes>
          <Route path='marvel' element={<MarvelScreen />} />
          <Route path='dc' element={<DcScreen />} />
          <Route path='search' element={<SearchScreen />} />
          <Route path='hero' element={<HeroScreen />} />
          <Route path='/' element={<MarvelScreen />} />
        </Routes>
      </div>
    </>
  )
}
