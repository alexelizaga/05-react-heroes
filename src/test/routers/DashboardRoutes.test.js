import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { AuthContext } from '../../auth/authContext';
import { DashboardRoutes } from "../../routers/DashboardRoutes";

describe('Test <DashboardRoutes />', () => {

  const contextValue = {
    user: {
      logged: true,
      name: 'Alex'
    }
  };

  test('should render correctly', () => {
    const view = render(
      <AuthContext.Provider value={ contextValue }>
        <MemoryRouter>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    );
    expect(view.baseElement).toMatchSnapshot();

    const userName = screen.queryByTestId('MyNavbarUserName');
    expect(userName.textContent.trim()).toBe(contextValue.user.name);
  });

});