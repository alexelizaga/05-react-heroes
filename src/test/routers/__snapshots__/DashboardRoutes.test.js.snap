// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <DashboardRoutes /> should render correctly 1`] = `
<body>
  <div>
    <nav
      class="navbar navbar-expand-sm navbar-dark bg-dark"
      data-testid="MyNavbar"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Asociaciones
        </a>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            class="me-auto navbar-nav"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="/marvel"
              href="/marvel"
            >
              Marvel
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="/dc"
              href="/dc"
            >
              DC
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="/search"
              href="/search"
            >
              Search
            </a>
          </div>
        </div>
        <div
          class="justify-content-end navbar-collapse collapse"
        >
          <span
            class="text-info navbar-text"
            data-testid="MyNavbarUserName"
          >
            Alex
          </span>
          <div
            class="navbar-nav"
          >
            <button
              class="btn btn-dark"
              type="button"
            >
              Logout
            </button>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="container"
    >
      <div>
        <h1
          data-testid="MarvelScreenTitle"
        >
          Marvel
        </h1>
        <hr />
        <div
          class="row row-cols-6 row-cols-md-4 row-cols-lg-3 d-flex align-items-stretch"
        >
          <div
            class="d-flex p-1"
          >
            <div
              class="card animate__animated animate__fadeIn"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Spider Man"
                    class="card-img-top"
                    src="/assets/marvel-spider.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Spider Man
                    </h5>
                    <p
                      class="card-text"
                    >
                      Peter Parker
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                        Amazing Fantasy #15
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-spider"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex p-1"
          >
            <div
              class="card animate__animated animate__fadeIn"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Captain America"
                    class="card-img-top"
                    src="/assets/marvel-captain.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Captain America
                    </h5>
                    <p
                      class="card-text"
                    >
                      Steve Rogers
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                        Captain America Comics #1
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-captain"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex p-1"
          >
            <div
              class="card animate__animated animate__fadeIn"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Iron Man"
                    class="card-img-top"
                    src="/assets/marvel-iron.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Iron Man
                    </h5>
                    <p
                      class="card-text"
                    >
                      Tony Stark
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                        Tales of Suspense #39
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-iron"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex p-1"
          >
            <div
              class="card animate__animated animate__fadeIn"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Thor"
                    class="card-img-top"
                    src="/assets/marvel-thor.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Thor
                    </h5>
                    <p
                      class="card-text"
                    >
                      Thor Odinson
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                        Journey into Myster #83
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-thor"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex p-1"
          >
            <div
              class="card animate__animated animate__fadeIn"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Hulk"
                    class="card-img-top"
                    src="/assets/marvel-hulk.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Hulk
                    </h5>
                    <p
                      class="card-text"
                    >
                      Bruce Banner
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                        The Incredible Hulk #1
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-hulk"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex p-1"
          >
            <div
              class="card animate__animated animate__fadeIn"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Wolverine"
                    class="card-img-top"
                    src="/assets/marvel-wolverine.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Wolverine
                    </h5>
                    <p
                      class="card-text"
                    >
                      James Howlett
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                        The Incredible Hulk #180
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-wolverine"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex p-1"
          >
            <div
              class="card animate__animated animate__fadeIn"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Daredevil"
                    class="card-img-top"
                    src="/assets/marvel-daredevil.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Daredevil
                    </h5>
                    <p
                      class="card-text"
                    >
                      Matthew Michael Murdock
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                        Daredevil #1
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-daredevil"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex p-1"
          >
            <div
              class="card animate__animated animate__fadeIn"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Hawkeye"
                    class="card-img-top"
                    src="/assets/marvel-hawkeye.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Hawkeye
                    </h5>
                    <p
                      class="card-text"
                    >
                      Clinton Francis Barton
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                        Tales of Suspense #57
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-hawkeye"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex p-1"
          >
            <div
              class="card animate__animated animate__fadeIn"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Cyclops"
                    class="card-img-top"
                    src="/assets/marvel-cyclops.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Cyclops
                    </h5>
                    <p
                      class="card-text"
                    >
                      Scott Summers
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                        X-Men #1
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-cyclops"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="d-flex p-1"
          >
            <div
              class="card animate__animated animate__fadeIn"
            >
              <div
                class="row no-gutters"
              >
                <div
                  class="col-4"
                >
                  <img
                    alt="Silver Surfer"
                    class="card-img-top"
                    src="/assets/marvel-silver.jpg"
                  />
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="card-body"
                  >
                    <h5
                      class="card-title"
                    >
                      Silver Surfer
                    </h5>
                    <p
                      class="card-text"
                    >
                      Norrin Radd
                    </p>
                    <p
                      class="card-text"
                    >
                      <small
                        class="text-muted"
                      >
                        The Fantastic Four #48
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-silver"
                    >
                      Más...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
